src/head_jump.h

```diff
@@ -93,20 +93,17 @@ extern void __orig___fentry__(void);
 #define JUMP_REMOVE_FUNC(func) 	\
 	memcpy((unsigned char *)__orig_##func, store_orig_##func, HEAD_LEN)
 
-static inline void do_write_cr0(unsigned long val)
+static inline void disable_write_protect(void)
 {
+	unsigned long val = read_cr0() & (~X86_CR0_WP);
 	asm volatile("mov %0,%%cr0": "+r" (val) : : "memory");
 }
 
-/* Must be used in stop machine context */
-#define JUMP_OPERATION(ops) do { 	\
-		unsigned long cr0;      \
-					\
-		cr0 = read_cr0();       \
-		do_write_cr0(cr0 & 0xfffeffff);    \
-		jump_##ops();		\
-		do_write_cr0(cr0);         \
-	} while(0)
+static inline void enable_write_protect(void)
+{
+	unsigned long val = read_cr0() | X86_CR0_WP;
```

Comment:
I think here should restore the CR0_WP bit instead of setting this bit explicitly, because this bit may have been disabled by others. The following way should be better:
1. store the orignal bits of CR0;
2. disable the CR0_WP;
3. restore the CR0;

Reply:
If other thread can interrupt CR0 configuration, the restore  mechanism doesn't work too.

store orig value1
disable WP bit
             <-------  write new value2 by other thread；
restore CR0 

What do you expect? value1 or value2? 

Actually, in stop machine context, nobody can interrupt CR0 configure process.